### PR TITLE
JAVA-2324 & JAVA-2325: Fixes for primitive types in the mapper

### PR DESCRIPTION
**Pull request by adutra**

### CODE REVIEW

File: integration-tests/src/test/java/com/datastax/oss/driver/mapper/PrimitivesIT.java [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+  private static TestMapper mapper;
+
+  @BeforeClass
+  public static void setup() {
+    CqlSession session = sessionRule.session();
+    session.execute(
+        SimpleStatement.builder(
+                "CREATE TABLE primitives_entity("
+                    + "id int PRIMARY KEY, "
```
Comment (tomekl007) [inline integration-tests/src/test/java/com/datastax/oss/driver/mapper/PrimitivesIT.java:64]: maybe you can add a similar test for wrapper types (`Integer`, `Load`, ...)?

Comment (adutra) [reply integration-tests/src/test/java/com/datastax/oss/driver/mapper/PrimitivesIT.java:64]: I don't think it's necessary, they are just normal objects so it's guaranteed that they will be treated exactly like any other type (`String`, `BigDecimal`...)

**Review by tomekl007 — APPROVED** [review]

File: mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java [hunk 2]
```diff
@@ -102,21 +102,36 @@ public EntityDefinition getDefinition(TypeElement classElement) {
...
         TypeMirror typeMirror = getMethod.getReturnType();
         if (typeMirror.getKind() == TypeKind.VOID) {
           continue;
         }
-        String propertyName = Introspector.decapitalize(getMethodName.substring(3));
+
+        String getMethodName = getMethod.getSimpleName().toString();
+        boolean regularGetterName = getMethodName.startsWith("get");
+        boolean booleanGetterName =
+            getMethodName.startsWith("is") && typeMirror.getKind() == TypeKind.BOOLEAN;
```
Comment (olim7t) [inline mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java]: We should also allow `Boolean.class`. I know the JavaBeans spec only mentions `boolean`, but we're not strictly adhering to it.

**Review by olim7t — APPROVED** [review]
LGTM.

Usual request to avoid "wishful" commit messages:
```
JAVA-2324: Add support for primitive shorts in mapper
JAVA-2325: Allow "is" prefix for boolean getters in mapped entities
```

**Comment by adutra** [comment]
> Usual request to avoid "wishful" commit messages:

OK, but I wanted to point out that imperative style imo does not work very well for bugs: `Add support for primitive shorts in mapper` makes it sound less like a bug and more like a new feature.

**Pull request merged**